pkg/skaffold/build/ko/build_test.go

```diff
@@ -0,0 +1,153 @@
+			ref:                 "gcr.io/project-id/example.com/myapp:myTag",
+			imageID:             "myTag",
+			pushImages:          true,
+			importpath:          "ko://example.com/myapp",
+			imageNameFromConfig: "ko://example.com/myapp",
+		},
+		{
+			description:         "workspace is not cwd",
+			ref:                 "gcr.io/project-id/example.com/test-app3:myTag",
+			imageID:             "imageID3",
+			pushImages:          false,
+			importpath:          "ko://github.com/GoogleContainerTools/skaffold/pkg/skaffold/build/docker",
+			imageNameFromConfig: "test-app3",
+			workspace:           "../docker",
+		},
+		{
+			description:         "ko import path image name and workspace is not cwd",
+			ref:                 "gcr.io/project-id/example.com/test-app4:myTag",
+			imageID:             "imageID4",
+			pushImages:          false,
+			importpath:          "ko://github.com/GoogleContainerTools/skaffold/pkg/skaffold/build/docker",
+			imageNameFromConfig: "ko://github.com/GoogleContainerTools/skaffold/pkg/skaffold/build/docker",
+			workspace:           "../docker",
+		},
+		{
+			description:         "ko import path image name and workspace is not cwd and import path is subdirectory of cwd",
+			ref:                 "gcr.io/project-id/example.com/test-app5:myTag",
+			imageID:             "imageID5",
+			pushImages:          false,
+			importpath:          "ko://github.com/GoogleContainerTools/skaffold/pkg/skaffold/build/docker",
+			imageNameFromConfig: "ko://github.com/GoogleContainerTools/skaffold/pkg/skaffold/build/docker",
+			workspace:           "..",
+		},
+	}
+	for _, test := range tests {
+		testutil.Run(t, test.description, func(t *testutil.T) {
+			b := stubKoArtifactBuilder(test.ref, test.imageID, test.pushImages, test.importpath)
+
+			artifact := &latestV1.Artifact{
+				ArtifactType: latestV1.ArtifactType{
+					KoArtifact: &latestV1.KoArtifact{},
+				},
+				Dependencies: []*latestV1.ArtifactDependency{},
+				ImageName:    test.imageNameFromConfig,
+				Workspace:    test.workspace,
+			}
+
+			var outBuffer bytes.Buffer
+			gotImageID, err := b.Build(context.TODO(), &outBuffer, artifact, test.ref)
+			t.CheckNoError(err)
+			if gotImageID != test.imageID {
+				t.Errorf("got image ID %s, wanted %s", gotImageID, test.imageID)
+			}
+			imageNameOut := strings.TrimSuffix(outBuffer.String(), "\n")
+			if imageNameOut != test.ref {
+				t.Errorf("image name output was %q, wanted %q", imageNameOut, test.ref)
+			}
+		})
+	}
+}
+
+func stubKoArtifactBuilder(ref string, imageID string, pushImages bool, importpath string) *Builder {
+	api := (&testutil.FakeAPIClient{}).Add(ref, imageID)
+	localDocker := fakeLocalDockerDaemon(api)
+	b := NewArtifactBuilder(localDocker, pushImages)
+
+	// Fake implementation of the `publishImages` function.
+	b.publishImages = func(_ context.Context, _ []string, _ publish.Interface, _ build.Interface) (map[string]name.Reference, error) {
```

Comment:
It's not clear to me what this is actually returning since this stub is ignoring its arguments.  Should it be using the importpath to populate the return results?  A comment would be helpful.

Reply:
I'll add a comment 👍